File: app/src/main/java/doit/study/droid/data/TagDao.java
package doit.study.droid.data;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import java.util.List;

@Dao
public interface TagDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(List<Tag> tags);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(Tag tag);

    @Query("SELECT * FROM tags")
    List<Tag> getAllTags();
}

File: app/src/main/java/doit/study/droid/data/local/dao/TagDao.kt
package doit.study.droid.data.local.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import doit.study.droid.data.local.entity.QuestionTagJoin
import doit.study.droid.data.local.entity.Tag

@Dao
interface TagDao {
    @Query("select * from Tag")
    suspend fun getTags(): List<Tag>

    @Query("select * from Tag where id = :id")
    suspend fun getTag(id: Int): Tag

    @Query("select * from Tag where selected = :isSelected")
    suspend fun getTagBySelection(isSelected: Boolean): List<Tag>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertTag(tag: Tag): Long

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertOrReplaceTag(vararg tags: Tag): List<Long>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertQuestionTagJoin(questionTagJoin: QuestionTagJoin)
}